modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java

```diff
@@ -1648,7 +1640,7 @@ private ObjectDefinition _publishObjectDefinition(
 		int count = _objectFieldPersistence.countByODI_S(
 			objectDefinition.getObjectDefinitionId(), false);
 
-		if (count == 0) {
+		if ((count == 0) && !objectDefinition.isSystem()) {
```

Comment:
Will there be an exception for trying to publish a system object definition?
Let's ask the product team.

Reply:
In my opinion it must have at least one custom field or one system field that is not metadata.
cc. @marcelabc @gabrielwas

Reply:
Agree :+1:

Reply:
To do: add a verification/validation in method "publish"